### PR TITLE
Migrate azure monitor metrics exporter from previous repo

### PR DESCRIPTION
**Pull request by lzchen**
From https://github.com/microsoft/opentelemetry-azure-monitor-python

### CODE REVIEW

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/azure/opentelemetry/exporter/azuremonitor/export/metrics/_exporter.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    )
+
+    properties = {}
+    for label_tuple in export_record.labels:
+        properties[label_tuple[0]] = label_tuple[1]
+    data = MetricsData(metrics=[data_point], properties=properties)
+    envelope.data = MonitorBase(base_data=data, base_type="MetricData")
+    return envelope
+
+# def standard_metrics_processor(envelope):
```
Comment (hectorhdzg) [inline sdk/monitor/azure-opentelemetry-exporter-azuremonitor/azure/opentelemetry/exporter/azuremonitor/export/metrics/_exporter.py:144]: Just delete this commented code, most likely standard metrics will end as a separate package after the discussion we had with Josh.

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/tests/metrics/test_metrics.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        self.assertEqual(envelope.data.base_data.metrics[0].name, "testname")
+        self.assertEqual(envelope.data.base_data.metrics[0].value, 400)
+        self.assertEqual(envelope.data.base_data.metrics[0].min, 100)
+        self.assertEqual(envelope.data.base_data.metrics[0].max, 300)
+        self.assertEqual(envelope.data.base_data.metrics[0].count, 2)
+        self.assertEqual(
+            envelope.data.base_data.properties["environment"], "staging"
+        )
+
+    # def test_standard_metrics_processor(self):
```
Comment (hectorhdzg) [inline sdk/monitor/azure-opentelemetry-exporter-azuremonitor/tests/metrics/test_metrics.py:262]: Same here, this test most likely would be in a different file

**Review by hectorhdzg — APPROVED** [review]
LGTM

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/azure/opentelemetry/exporter/azuremonitor/export/metrics/_exporter.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+class AzureMonitorMetricsExporter(BaseExporter, MetricsExporter):
+    """Azure Monitor metrics exporter for OpenTelemetry.
+
+    :param options: Exporter configuration options.
+    :type options: ~azure.opentelemetry.exporter.azuremonitor.options.ExporterOptions
+    """
+
+    def export(
+        self, export_records: Sequence[ExportRecord]
+    ) -> MetricsExportResult:
```
Comment (rakshith91) [inline sdk/monitor/azure-opentelemetry-exporter-azuremonitor/azure/opentelemetry/exporter/azuremonitor/export/metrics/_exporter.py:45]: add **kwargs?

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/samples/metrics/observer.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import psutil
```
Comment (rakshith91) [inline sdk/monitor/azure-opentelemetry-exporter-azuremonitor/samples/metrics/observer.py:3]: All the samples need docstrings - please create an issue to make sure we have it tracked :)

**Review by rakshith91 — APPROVED** [review]

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/azure/opentelemetry/exporter/azuremonitor/export/metrics/_exporter.py (hunk 3 above)
Comment (lzchen) [reply sdk/monitor/azure-opentelemetry-exporter-azuremonitor/azure/opentelemetry/exporter/azuremonitor/export/metrics/_exporter.py:45]: `export` is never called by the user so I do not believe there is a need for `**kwargs`

File: sdk/monitor/azure-opentelemetry-exporter-azuremonitor/samples/metrics/observer.py (hunk 4 above)
Comment (lzchen) [reply sdk/monitor/azure-opentelemetry-exporter-azuremonitor/samples/metrics/observer.py:3]: Will create an issue.

Comment (lzchen) [reply sdk/monitor/azure-opentelemetry-exporter-azuremonitor/samples/metrics/observer.py:3]: https://github.com/Azure/azure-sdk-for-python/issues/16382

**Comment by lzchen** [comment]
Closing for this PR to the feature branch instead: https://github.com/Azure/azure-sdk-for-python/pull/16383

**Pull request closed without merging**